vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors_moe.py

```diff
@@ -96,8 +101,14 @@ def get_moe_method(
 class CompressedTensorsW4A4MoeMethod(CompressedTensorsMoEMethod):
 
     def __init__(self):
-        self.use_marlin = not cutlass_fp4_supported()
+        from vllm.model_executor.layers.quantization.utils.nvfp4_support import (  # noqa: E501
+            detect_nvfp4_support)
+        _nvfp4 = detect_nvfp4_support(self.__class__.__name__, logger)
+        self.cutlass_nvfp4_supported = _nvfp4.cutlass_supported
+        self.allow_flashinfer_cutlass = _nvfp4.allow_flashinfer_cutlass
+        self.use_marlin = _nvfp4.use_marlin
```

Comment:
Could we just pass in self to the function? Then the function could take care of assigning these variables as well as getting the class name. 
Also, I don't see why we should pass in the logger here and the function name should include moe if this is moe specific

Reply:
Yeah, the origin version is detect_nvfp4_support(obj, logger), and we set everything inside the function, like `obj. allow_flashinfer_cutlass = True`. But after consideration, this won't be a good practice since class attributes should be set inside class so that it is easy to refer and read.

Reply:
Logger and function name should include moe fixed, thanks!